packages/integration-tests/src/tests/api/oidc/token-exchange.test.ts

```diff
@@ -134,5 +137,119 @@ describe('Token Exchange', () => {
 
       await deleteApplication(thirdPartyApplication.id);
     });
+
+    it('should filter out non-oidc scopes', async () => {
+      const { subjectToken } = await createSubjectToken(userId);
+
+      const body = await oidcApi
+        .post('token', {
+          headers: formUrlEncodedHeaders,
+          body: new URLSearchParams({
+            client_id: applicationId,
+            grant_type: GrantType.TokenExchange,
+            subject_token: subjectToken,
+            subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+            scope: [UserScope.Profile, 'non-oidc-scope'].join(' '),
+          }),
+        })
+        .json();
+
+      expect(body).toHaveProperty('access_token');
+      expect(body).toHaveProperty('token_type', 'Bearer');
+      expect(body).toHaveProperty('expires_in');
+      expect(body).toHaveProperty('scope', UserScope.Profile);
+    });
+  });
+
+  describe('get access token for organization', () => {
+    const scopeName = `read:${randomString()}`;
+
+    /* eslint-disable @silverhand/fp/no-let */
+    let testApiScopeId: string;
+    let testOrganizationId: string;
+    /* eslint-enable @silverhand/fp/no-let */
+
+    const organizationApi = new OrganizationApiTest();
+
+    /* eslint-disable @silverhand/fp/no-mutation */
+    beforeAll(async () => {
+      const organization = await organizationApi.create({ name: 'org1' });
+      testOrganizationId = organization.id;
+      await organizationApi.addUsers(testOrganizationId, [userId]);
+
+      const scope = await organizationApi.scopeApi.create({ name: scopeName });
+      testApiScopeId = scope.id;
+
+      const role = await organizationApi.roleApi.create({ name: `role1:${randomString()}` });
+      await organizationApi.roleApi.addScopes(role.id, [scope.id]);
+      await organizationApi.addUserRoles(testOrganizationId, userId, [role.id]);
+    });
+    /* eslint-enable @silverhand/fp/no-mutation */
+
+    afterAll(async () => {
+      await organizationApi.cleanUp();
+    });
+
+    it('should be able to get access token for organization with correct scopes', async () => {
+      const { subjectToken } = await createSubjectToken(userId);
+
+      const { access_token } = await oidcApi
+        .post('token', {
+          headers: formUrlEncodedHeaders,
+          body: new URLSearchParams({
+            client_id: applicationId,
+            grant_type: GrantType.TokenExchange,
+            subject_token: subjectToken,
+            subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+            organization_id: testOrganizationId,
+            scope: scopeName,
```

Comment:
will we add `resource` and `resource` + `organization_id` test cases later?

Reply:
Yes, add LOG-9521 to track.